api/src/main/java/jakarta/data/repository/Find.java

```diff
@@ -27,48 +27,49 @@
- *     must match, or the entity attribute names must be specified by the
- *     {@link Select} annotation.</li>
+ * <li>If the method return involves an entity type, according to the rules
+ *     specified below, the return type determines the queried entity type.</li>
+ * <li>Otherwise, if the method return type does not involve an entity type,
+ *     the queried entity type is the {@linkplain #value entity class explicitly
+ *     specified by the annotation}, if any, or the primary entity type of the
+ *     repository.</li>
  * </ul>
  *
- * <p>When returning a single entity attribute or a subset of entity attributes,
- * the attribute types must be single-valued basic types, not multiple-valued
- * types such as collections, arrays, or associations.</p>
- *
- * <p>If the method return type is not an entity type, the queried entity type
- * is determined by:</p>
+ * <p>The return type of the annotated repository method must involve either:</p>
  * <ul>
- * <li>the {@link #value entity class} explicitly specified by the annotation,
- *     if any, or
- * <li>the primary entity type of the repository, if any.
+ * <li>the queried entity type itself,</li>
+ * <li>the type of the attribute of the queried entity type specified by the
+ *     {@link Select} annotation, or</li>
+ * <li>a Java record type packaging the entity attributes returned by the
+ *     query. The names and types of the record components must match the
+ *     names and types of attributes of the queried entity, or the attribute
+ *     names must be explicitly specified by the {@link Select} annotation.</li>
  * </ul>
+ * <p>When the return type is not the queried entity type, every returned
+ * attribute must be a basic attribute.</p>
  *
- * <p>Each parameter of the annotated method must either:
- * </p>
+ * <p>Each parameter of the annotated method must either:</p>
  * <ul>
- * <li>have exactly the same type and name (the parameter name in the Java source, or a name assigned by {@link By @By})
- *     as an attribute of the entity class, or</li>
- * <li>be of type {@link jakarta.data.Limit}, {@link jakarta.data.Sort}, {@link jakarta.data.Order}, or
- *     {@link jakarta.data.page.PageRequest}.</li>
+ * <li>have exactly the same type and name (the parameter name in the Java source,
+ *     or a name assigned by {@link By @By}) as an attribute of the entity class,
+ *     or</li>
+ * <li>be of type {@link jakarta.data.Limit}, {@link jakarta.data.Sort},
+ *     {@link jakarta.data.Order}, or {@link jakarta.data.page.PageRequest}.</li>
  * </ul>
- * <p>The query is inferred from the method parameters which match attributes of the entity.
- * </p>
- * <p>There is no specific naming convention for methods annotated with {@code @Find}; they may be named arbitrarily,
- * and their names do not carry any semantic meaning defined by the Jakarta Data specification.
- * </p>
+ * <p>The query is inferred from the method parameters which match attributes of
+ *  entity.</p>
```

Comment:
A word was lost when reformatting:
```suggestion
 * <p>The query is inferred from the method parameters which match attributes of
 * the entity.</p>
```